### ISSUE TITLE
Tutorials: Use wcag22.json for "Related WCAG resources"

### ISSUE THREAD

**Issue by remibetin** [w3c/wai-website#2046]
### Resource URL

https://www.w3.org/WAI/tutorials/

### Description

Tutorial pages include links to related success criteria and techniques.
Example: https://www.w3.org/WAI/tutorials/page-structure/headings/

We currently use:
- [wai-website/wcag-legacy.yml](https://github.com/w3c/wai-website/blob/main/_data/wcag-legacy.yml) for SC content
- [wai-website/techniques-legacy.yml](https://github.com/w3c/wai-website/blob/main/_data/techniques-legacy.yml) for Techniques content

We should leverage the more recent [wai-website-theme/wcag22.json](https://github.com/w3c/wai-website-theme/blob/main/_data/wcag22.json) file instead.

**Note:** We'll have to figure out how to display SCs with a long description. Currently, this is not handled consistently:
- For some SCs, only the first part has been preserved in `wai-website/wcag-legacy.yml`
  Example: 
  > All non-text content that is presented to the user has a text alternative that serves the equivalent purpose, except for the situations listed[…].
- For others, the full description has been kept. 
  Example: 
  > The visual presentation of text and images of text has a contrast ratio of at least 4.5:1, except for the following:<ul><li><strong>Large Text: </strong> Large-scale text and images of large-scale text have a contrast ratio of at least 3:1;</li><li><strong>Incidental: </strong> Text or images of text that are part of an inactive user interface component, that are pure decoration, that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.</li><li><strong>Logotypes: </strong> Text that is part of a logo or brand name has no minimum contrast requirement.</li></ul>

We should also directly use the new success criteria IDs ([related discussion](https://github.com/w3c/wai-website-theme/pull/201#discussion_r3305567732)).

**Comment by salheyuiop-eng** [comment]
السلام عليكم

**Comment by kfranqueiro** [comment]
I looked into the `wcag-legacy.yml` usage first. Techniques will be more complex and may warrant more logic in `_update-wcag-json.mjs`, as there is no direct equivalent to the flattened data that is currently used; associated techniques are structured hierarchically under each SC in `wcag22.json`.

If you want to preview this for yourself, update your wai-website Gemfile to point to the [kgf-tutorials-new-data](https://github.com/w3c/wai-website-theme/compare/kgf-tutorials-new-data) branch for wai-website-theme.

> **Note:** We'll have to figure out how to display SCs with a long description. Currently, this is not handled consistently:
>      
> * For some SCs, only the first part has been preserved in `wai-website/wcag-legacy.yml`
> * For others, the full description has been kept.

I found 2 cases that were specially truncated in the legacy data and are visible in the tutorials:

- 1.1.1 Non-text Content has "below" replaced with "[...]" and the subsequent definition list is removed
- 3.3.4 Error Prevention replaces the definition list with just the 3 terms inline ("Reversible, Checked, Confirmed")

With the exception of those (which the logic below will fully expand), I was able to get equivalent treatment programmatically using `wcag22.json` with the following logic:

```liquid
{%- assign titleEndChar = sc.title | slice: -1 -%}
{%- if titleEndChar == ":" or sc.title contains "below." %} {{sc.content}} {% else %} {{sc.title}} {% endif %} (Level&nbsp;{{sc.level}})
```

This will show the full HTML content for any SC whose first paragraph ends with a colon, and also includes the 1.1.1 case which ends with "below." For the rest of the SCs, it will only show the first paragraph.

However, the full-HTML representation will not be identical to the old; it will be much more similar to what's seen in the Recommendation itself. Notably:

- The first sentence will always be in its own paragraph rather than on the same line
- Things that were bulleted lists in the old data are definition lists in the new data, because that's what they are in the Recommendation; this arguably formats them more consistently than before, e.g. look at the two lists currently in [Multi-page Forms](https://www.w3.org/WAI/tutorials/forms/multi-page/#related)
- Notes from the SC will be included

Is this acceptable? If not, we'd have to script any tweaks; it already feels odd that manual tweaks were previously made to a seemingly-central data file for this specific use case.

Tutorial pages containing relevant differences:

- `images/` (1.1.1, 1.4.5)
- `forms/validation/` (3.3.4)
- `forms/multi-page/` (2.2.1, 2.2.2)
- `carousels/` (2.2.2)
- `carousels/animations/` (2.2.2)
- `carousels/styling/` (1.4.3, 2.5.5 which wasn't in legacy data)

> We should also directly use the new success criteria IDs ([related discussion](https://github.com/w3c/wai-website-theme/pull/201#discussion_r3305567732)).

This should not be a problem, as we already discovered (in the related thread) that either the new or old IDs work with the Quickref, and it's trivial to get the new IDs out of `wcag22.json`. Any links I've tested so far with new IDs have worked.

**Comment by remibetin** [comment]
Thanks much for your investigation and work on this, @kfranqueiro! The list of tutorial pages containing relevant differences was also very helpful.

> Is this acceptable? If not, we'd have to script any tweaks; it already feels odd that manual tweaks were previously made to a seemingly-central data file for this specific use case.

Unfortunately, I don't think it is: it really clutters the list of links. This is particularly noticeable on the Multi-page Forms page.

I'd also like to avoid manual tweaks or partial display of the success criteria content. Both are tricky in their own way. Actually, I even wonder if this use case warrants displaying the success criteria content. The described purpose of this section is to provide pointers to WCAG resources that were combined to develop a given tutorial.

For reference, the sentence that introduces the lists of success criteria and techniques:

> This page combined the following WCAG success criteria and techniques from different conformance levels:

In this context, I don't feel the success criteria content is absolutely necessary.

Could you try a version that only displays the SC number, handle, and conformance level? Preferably on a different branch, so we can compare the two.

**Comment by kfranqueiro** [comment]
I wasn't sure how helpful 2 separate branches here would be given they'd still need to be checked locally (as the Netlify builds on the theme repo would not show these pages), so I have pushed subsets of 2 builds to Codeberg, since the legacy version of their pages infrastructure allows referencing different branches from the same repo:

- [With descriptions](https://kgf.codeberg.page/wai-website-tutorial-data-previews/@with-desc/tutorials/)
- [Without descriptions](https://kgf.codeberg.page/wai-website-tutorial-data-previews/@without-desc/tutorials/)

I have only included the tutorial pages, so anything else will 404. But for instance, you can compare the multi-form page [with](https://kgf.codeberg.page/wai-website-tutorial-data-previews/@with-desc/tutorials/forms/multi-page/) and [without](https://kgf.codeberg.page/wai-website-tutorial-data-previews/@without-desc/tutorials/forms/multi-page/) descriptions.

(RE clutter: to be clear, that was arguably already pretty bad for a few of them in the current live version, but the new data does make that worse.)

I also finished working on the data updates needed for techniques, so those links are also updated in those two builds. All of this is on my `kgf-tutorials-new-data` branch, including the removal of descriptions as a separate commit that I can remove if we don't want it.

The branch also contains an isolated commit bringing wcag22.json up to date with current docs, which I can PR separately ahead-of-time so that the PR for this will only introduce the new `techniques` section.

**Comment by remibetin** [comment]
@kfranqueiro

> I wasn't sure how helpful 2 separate branches here would be given they'd still need to be checked locally (as the Netlify builds on the theme repo would not show these pages)

For context, I sometimes use draft PRs on the `wai-website` repository to preview theme changes with Netlify. Mainly when I need others to take a quick look. The previews you've provided with Codeberg are just as good: thanks!

> I have only included the tutorial pages, so anything else will 404. But for instance, you can compare the multi-form page [with](https://kgf.codeberg.page/wai-website-tutorial-data-previews/@with-desc/tutorials/forms/multi-page/) and [without](https://kgf.codeberg.page/wai-website-tutorial-data-previews/@without-desc/tutorials/forms/multi-page/) descriptions.

I strongly prefer the version without descriptions. I've asked Becky and Shawn for input.

> (RE clutter: to be clear, that was arguably already pretty bad for a few of them in the current live version, but the new data does make that worse.)

+1

> I also finished working on the data updates needed for techniques, so those links are also updated in those two builds.

This looks good to me on the Multi-page forms page. Will look at a few more pages during the PR review.

**Comment by shawna-slh** [comment]
/me skims briefly...

I am fine with whatever you all decide. Good to get @becka11y 's perspective, too.

Somewhat tangential, though removing the text of the SC makes it more likely readers will follow links: I do wonder if it's best to link users to the QuickRef? Think of this use case: A user who jumps into a tutorials with little or no knowledge of WCAG and the different documents. They are comfortable with the tutorials. To learn more, they following one of the SC links. Currently that drops them in the middle of the QuickRef. **_OVERWHELMING_**
However, other options are not clearly better over all: standard or Understanding doc.

Anyway, I'm fine to leave y'all to decide on this, too.

**Comment by shawna-slh** [comment]
> Anyway, I'm fine to leave y'all to decide on this, too.

Including not taking the time to consider about this potential change.

**Comment by remibetin** [comment]
> Somewhat tangential, though removing the text of the SC makes it more likely readers will follow links: I do wonder if it's best to link users to the QuickRef? Think of this use case: A user who jumps into a tutorials with little or no knowledge of WCAG and the different documents. They are comfortable with the tutorials. To learn more, they following one of the SC links. Currently that drops them in the middle of the QuickRef. OVERWHELMING
However, other options are not clearly better over all: standard or Understanding doc.

Thanks for bringing this up! I wondered about that too and started a separate internal issue 685, so we can focus on the original scope here.

**Comment by becka11y** [comment]
Just focusing on whether on not to include the description with the SC listing, I prefer just the SC with no description.  If someone is new, I think the description can make it feel a bit more overwhelming.  Yes, it provides exceptions and more nuance but a "newbie" wouldn't be ready for that. Someone more familiar with WCAG  will likely be more comfortable to following the link to get the additional details. And, the page just feels cleaner and easier to follow without the description. 

I'm not sure where to look for internal issue 685? I do have opinions on which document(s) to link to ;)

**Comment by remibetin** [comment]
Thanks @becka11y. The [merged PR](https://github.com/w3c/wai-website-theme/pull/213) displays the SCs without descriptions. 

> I do have opinions on which document(s) to link to ;)

I took away from separate discussions with you and with @kfranqueiro yesterday that we all leaned toward linking to Understanding documents instead of QuickRef. The change has been included in the [merged PR](https://github.com/w3c/wai-website-theme/pull/213). Thanks!
